src/UI/Actions.hs

```diff
@@ -388,8 +390,22 @@ removeTags ts =
 reloadList :: Action 'BrowseThreads AppState
 reloadList = Action "reload list of threads" applySearch
 
+selectNextUnread :: Action 'BrowseMail AppState
+selectNextUnread =
+  Action { _aDescription = "select next unread"
+         , _aAction = (\s -> let vec = (view (asMailIndex . miListOfMails . L.listElementsL) s)
+                                 cur = (view (asMailIndex . miListOfMails . L.listSelectedL) s) <|> Just 0
+                                 fx m = Notmuch.hasTag (view (asConfig . confNotmuch . nmNewTag) s) m
+                                 next = nextUnread vec cur fx
+                             in pure $ over (asMailIndex . miListOfMails) (L.listMoveTo (fromMaybe 0 next)) s)
+         }
+
 -- Function definitions for actions
 --
+-- | Return the next unread item or default to the currently selected.
+nextUnread :: Vector.Vector a -> Maybe Int -> (a -> Bool) -> Maybe Int
+nextUnread vec cur fx = Vector.findIndex fx (Vector.drop ((fromMaybe 0 cur) + 1) vec) <|> cur
```

Comment:
Seems like you should be adding the current offset to the result of `findIndex`?

Reply:
```haskell
findIndexWithOffset i f = fmap (i+) . V.findIndex f . V.drop i

seekIndex i f = fromMaybe i . findIndexWithOffset (i + 1) f
```

Reply:
Alright got this in. I'll try to find a property to maybe test this.